### PR TITLE
support launching as a preview

### PR DESCRIPTION
**Pull request by nathanstitt**
Previews do not record launch or viewed state on qualtrics

image

### CODE REVIEW

File: backend/app/services/launch_pad.rb [hunk 1]
```diff
@@ -7,26 +7,20 @@ def initialize(study_id:, user_id:)
     @user_id = user_id
   end
 
-  def launch
+  def launch_url(preview: false)
     raise(LaunchError, 'This study is not open.') unless study.open?
```
Comment (reedstrm) [inline backend/app/services/launch_pad.rb:11]: Shouldn't we allow researchers to preview studies that are not yet open?

File: backend/app/services/launch_pad.rb [hunk 2]
```diff
@@ -7,26 +7,17 @@ def initialize(study_id:, user_id:)
...
-        else
-          raise "Unsupported stage type: '#{launch_config[:type]}'"
-        end
-
-      launcher.new(config: launch_config, user_id: user_id).url
-    end
-
-    url || raise('An error occurred when building a launch url')
+    if preview
+      study.stages.order(:order).first.launcher(user_id).url + '&Q_CHL=preview&Q_SurveyVersionID=current'
```
Comment (reedstrm) [inline backend/app/services/launch_pad.rb]: Hmm, this is a Qualtrix-specific method of making a preview URL - knowledge of qualtrix-ness is buried in the launcher, so I think we should extend that model to have a preview_url component, instead?

**Review by reedstrm — CHANGES_REQUESTED** [review]
Huh, thought this preview work was going to be a frontend thing. From what I can see, all the frontend work is changing implementation of some bit from React to JSX, and lint work?

Two Qs.

File: backend/app/services/launch_pad.rb (hunk 1 above)
Comment (nathanstitt) [reply backend/app/services/launch_pad.rb:11]: yes, I suppose we might as well allow that. shouldn't hurt anything and I can see where it'd be useful

File: backend/app/services/launch_pad.rb (hunk 2 above)
Comment (nathanstitt) [reply backend/app/services/launch_pad.rb]: good catch, yep this should live in qualtrics launcher.  other launchers might not support preview and could raise an exception or something

**Review by reedstrm — APPROVED** [review]

**Pull request merged**